cabal-install/src/Distribution/Client/ScriptUtils.hs

```diff
@@ -125,7 +125,12 @@ import qualified Text.Parsec as P
 -- Two hashes will be the same as long as the absolute paths
 -- are the same.
 getScriptHash :: FilePath -> IO String
-getScriptHash script = showHashValue . hashValue . fromString <$> canonicalizePath script
+getScriptHash script
+  -- Base64 is shorter than Base16, which helps avoid long path issues on windows
+  -- but it can contain /'s which aren't valid in file paths so replace them with
+  -- %'s. Also drop unnecessary padding.
+  = map (\c -> if c == '/' then '%' else c) . dropWhileEnd (== '=')
```

Comment:
This is probably naive but, isn’t `%` used for environment vars on Windows? Wouldn’t there be a clash possibility then?

Reply:
According to https://en.wikipedia.org/wiki/Filename#In_Windows `%` is not special on Windows, other than when used in the shell it would have to be escaped, presumably, like `$` in Linux.